### PR TITLE
Check value is not null before creating setting.

### PR DESCRIPTION
**Pull request by hibooboo2**

### CODE REVIEW

**Comment by hibooboo2** [comment]
```
SQL [insert into `setting` (`name`, `value`) values (?, ?)]; Column 'value' cannot be null
	at org.jooq.impl.Utils.translate(Utils.java:1287) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:495) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:326) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.UpdatableRecordImpl.storeInsert0(UpdatableRecordImpl.java:230) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.UpdatableRecordImpl.access$100(UpdatableRecordImpl.java:90) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.UpdatableRecordImpl$2.operate(UpdatableRecordImpl.java:201) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:123) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.UpdatableRecordImpl.storeInsert(UpdatableRecordImpl.java:196) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.UpdatableRecordImpl.insert(UpdatableRecordImpl.java:144) ~[jooq-3.3.0.jar:na]
	at io.cattle.platform.object.impl.JooqObjectManager$1.executeNoResult(JooqObjectManager.java:80) ~[classes/:na]
	at io.cattle.platform.engine.idempotent.IdempotentExecutionNoReturn.execute(IdempotentExecutionNoReturn.java:7) ~[classes/:na]
	at io.cattle.platform.engine.idempotent.IdempotentExecutionNoReturn.execute(IdempotentExecutionNoReturn.java:3) ~[classes/:na]
	at io.cattle.platform.engine.idempotent.Idempotent.change(Idempotent.java:88) ~[classes/:na]
	at io.cattle.platform.object.impl.JooqObjectManager.insert(JooqObjectManager.java:77) ~[classes/:na]
	at io.cattle.platform.object.impl.AbstractObjectManager.create(AbstractObjectManager.java:62) ~[classes/:na]
	at io.cattle.platform.object.impl.AbstractObjectManager.create(AbstractObjectManager.java:72) ~[classes/:na]
	at io.cattle.platform.iaas.api.auth.SettingsUtils.changeSetting(SettingsUtils.java:22) ~[classes/:na]
	at io.cattle.platform.iaas.api.auth.integration.github.GithubConfigManager.updateCurrentConfig(GithubConfigManager.java:130) ~[classes/:na]
	at io.cattle.platform.iaas.api.auth.integration.github.GithubConfigManager.createInternal(GithubConfigManager.java:75) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractBaseResourceManager.create(AbstractBaseResourceManager.java:115) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.request.handler.ResourceManagerRequestHandler.generate(ResourceManagerRequestHandler.java:39) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.request.handler.AbstractResponseGenerator.handle(AbstractResponseGenerator.java:14) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.handle(DefaultReadWriteApiDelegate.java:27) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.write(DefaultReadWriteApiDelegate.java:22) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_71]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_71]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_71]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_71]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:96) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:260) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:94) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at com.sun.proxy.$Proxy43.write(Unknown Source) ~[na:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.ReadWriteApiHandler.handle(ReadWriteApiHandler.java:19) ~[classes/:na]
	at io.github.ibuildthecloud.gdapi.servlet.ApiRequestFilterDelegate.doFilter(ApiRequestFilterDelegate.java:86) ~[classes/:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter$1.runInContext(ApiRequestFilter.java:84) [classes/:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [classes/:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [classes/:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [classes/:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [classes/:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [classes/:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter.doFilter(ApiRequestFilter.java:77) [classes/:na]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:300) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [jetty-security-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_71]
Caused by: java.sql.SQLIntegrityConstraintViolationException: Column 'value' cannot be null
	at org.mariadb.jdbc.internal.SQLExceptionMapper.get(SQLExceptionMapper.java:132) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.internal.SQLExceptionMapper.throwException(SQLExceptionMapper.java:106) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.executeQueryEpilog(MySQLStatement.java:264) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.execute(MySQLStatement.java:288) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.executeUpdate(MySQLStatement.java:317) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLPreparedStatement.executeUpdate(MySQLPreparedStatement.java:156) ~[mariadb-java-client-1.1.7.jar:na]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105) ~[commons-dbcp-1.4.jar:1.4]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105) ~[commons-dbcp-1.4.jar:1.4]
	at org.jooq.tools.jdbc.DefaultPreparedStatement.executeUpdate(DefaultPreparedStatement.java:88) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractStoreQuery.execute(AbstractStoreQuery.java:346) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:322) ~[jooq-3.3.0.jar:na]
	... 66 common frames omitted
Caused by: org.mariadb.jdbc.internal.common.QueryException: Column 'value' cannot be null
	at org.mariadb.jdbc.internal.mysql.MySQLProtocol.getResult(MySQLProtocol.java:942) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.internal.mysql.MySQLProtocol.executeQuery(MySQLProtocol.java:991) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.execute(MySQLStatement.java:281) ~[mariadb-java-client-1.1.7.jar:na]
	... 73 common frames omitted

```

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/SettingsUtils.java [hunk 1]
```diff
@@ -16,9 +16,15 @@ public void changeSetting(String name, Object value) {
...
+
+        if (value == null) {
+            value = "";
+        }
+
         Setting setting = objectManager.findOne(Setting.class, "name", name);
+
         if (setting == null) {
-            objectManager.create(Setting.class, "name", name, "value", value);
+                objectManager.create(Setting.class, "name", name, "value", value);
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/SettingsUtils.java]: why the whitespace change?

Comment (hibooboo2) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/SettingsUtils.java]: Accidental. do you want me to fix it?

**Comment by cjellick** [comment]
LGTM

**Pull request merged**